Barotrauma/BarotraumaClient/ClientSource/Characters/CharacterNetworking.cs

```diff
@@ -220,7 +220,16 @@ public void ClientReadPosition(IReadMessage msg, float sendingTime)
                 keys[(int)InputType.Ragdoll].Held = ragdollInput;
                 keys[(int)InputType.Ragdoll].SetState(false, ragdollInput);
 
-                facingRight = msg.ReadBoolean();
+            }
+
+            facingRight = msg.ReadBoolean();
+
+            if (AnimController.IsFlipped == facingRight)
+            {
+                if((AnimController is HumanoidAnimController && !CanMove) || AnimController is FishAnimController)
+                {
+                    TryFlipCharacter();
+                }
```

Comment:
This seems like an incorrect way to handle this to me. The event we're reading here is not necessarily the latest one or received in order: that's why we add it to the `memState` buffer later in the method, and process that buffer in the proper order. I also feel this is not the correct place to be handling this - animation logic should be handled in the animation code, not a networking method that reads the remote state of the character.

I also find it unclear why we need to now add the facing to the buffer and handle it later, if it applies immediately.